src/txdb.cpp

```diff
@@ -41,35 +42,45 @@ struct CoinEntry {
 
 }
 
-CCoinsViewDB::CCoinsViewDB(fs::path ldb_path, size_t nCacheSize, bool fMemory, bool fWipe) : db(ldb_path, nCacheSize, fMemory, fWipe, true)
+CCoinsViewDB::CCoinsViewDB(fs::path ldb_path, size_t nCacheSize, bool fMemory, bool fWipe) :
+    m_ldb_path(ldb_path),
+    m_is_memory(fMemory),
```

Comment:
0fd9039ab3de00a0ac03b8fe0463f0c62a92dd43

This currently introduces a new compiler warning. The initializer list should be in the same order as they members are declared.

```
txdb.cpp:47:5: warning: field 'm_is_memory' will be initialized after field 'm_db' [-Wreorder]
    m_is_memory(fMemory),
```

Reply:
FWIW I can't reproduce this locally with either clang or gcc:

```
CC            = /usr/bin/ccache /usr/bin/clang
CFLAGS        = -g -O2
CPPFLAGS      =  -DDEBUG -DDEBUG_LOCKORDER  -I/home/james/src/bitcoin/db4/include/ -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS
CXX           = /usr/bin/ccache /usr/bin/clang++ -std=c++11
CXXFLAGS      =  -O0 -g3 -ftrapv  -Wstack-protector -fstack-protector-all     -std=c++11 -Wthread-safety-analysis -Werror

% clang++ --version
clang version 7.0.1-8 (tags/RELEASE_701/final)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin

% gcc --version
gcc (Debian 8.3.0-6) 8.3.0
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

Reply:
You just need to move `m_db` to be the first thing initialized, just as it's the first thing declared in txdb.h. Warning appears for me with clang 9.0 fwiw. cf https://en.cppreference.com/w/cpp/language/constructor#Initialization_order

Reply:
Fixed, thanks